src/openms/include/OpenMS/KERNEL/RangeManager.h

```diff
@@ -28,181 +28,574 @@
 // ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 //
 // --------------------------------------------------------------------------
-// $Maintainer: Timo Sachsenberg$
-// $Authors: Marc Sturm $
+// $Maintainer: Chris Bielow $
+// $Authors: Chris Bielow $
 // --------------------------------------------------------------------------
 
 #pragma once
 
-#include <OpenMS/DATASTRUCTURES/DRange.h>
+#include <OpenMS/config.h>
+#include <OpenMS/CONCEPT/Exception.h>
+
+#include <algorithm> // for min/max
+#include <cassert>
+#include <iosfwd>  // for std::ostream
+
 
 namespace OpenMS
 {
```

Comment:
can you add a few lines of comment (@brief ) so people understand the design decision.
blabla static polymorphism using visitor patterns blablaba maybe link to some stackoverflow or external resource